worker/machineactions/handleactions.go

```diff
@@ -40,6 +43,7 @@ func HandleAction(name string, params map[string]interface{}) (results map[strin
 func handleJujuRunAction(params map[string]interface{}) (results map[string]interface{}, err error) {
 	// The spec checks that the parameters are available so we don't need to check again here
 	command, _ := params["command"].(string)
+	logger.Debugf("juju run %q", command)
```

Comment:
Maybe trace? 

And what about potentially sensitive things on the command line?